docs/contributing/999-rfcs/2024-02-15-resource-pinning.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Resource Pinning
+description: Protection of resources from accidental deletion/moves
+---
+
+# Resource Pinning
+
+- **Author(s):**: @MarkMcCulloh
+- **Submission Date**: 2024-02-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-02-15
+
+Introduce a mechanism to protect resources from accidental deletion/moves in the logical application tree.
+
+## Requirements
+
+### Functional
```

Comment:
Is there a requirement to be able to control the policy? In CFN and TF there's a distinction between "disallow removal" and "leave orphan".

Reply:
I'm not sure what this means in the context of the logical tree.

In the context of CFN/TF, I considered platform-level handling out of scope for this RFC. Presumably this is the sort of metadata that can be added to the pinfile next to.

Reply:
Yes, sounds like the right direction to put this in the pinfile as additional attributes.

Might want to make sure the schema of this file allows it.

Reply:
> In CFN and TF there's a distinction between "disallow removal" and "leave orphan".

Does TF have "leave orphan"? I mean you can technically remove a tf resource from the json and the statefile but terraform itself doesn't have a any way to declare that intention AFAIK.

If TF doesn't have it, I think it's okay leaving out for now. We can add it to the pinfile but it would likely not be implemented in the first version.